pkg/cli/serve/serve.go

```diff
@@ -130,13 +175,54 @@ func (o *serveOptions) Run() error {
 		return err
 	}
 
-	// Initialize ocm sdk connection client
-	sdkclient, err := ocm.NewConnection().Build(viper.GetString(config.OcmURL),
-		viper.GetString(config.ClusterID),
-		viper.GetString(config.AccessToken))
-	if err != nil {
-		o.logger.WithError(err).Fatal("Can't initialise OCM sdk.Connection client")
-		return err
+	// Depending on whether the FleetMode is enabled or not, we need to initiate the OCM SDK connection accordingly
+	// If fleet mode is not enabled, we will fetch the cluster ID and access token to initiate connection with OCM
+	if !o.fleetMode {
+		sdkclient, err = ocm.NewConnection().Build(viper.GetString(config.OcmURL),
+			viper.GetString(config.ClusterID),
+			viper.GetString(config.AccessToken))
+		if err != nil {
+			o.logger.WithError(err).Fatal("Can't initialise OCM sdk.Connection client in non-fleet mode")
+			return err
+		}
+		o.logger.Info("Connection with OCM initialised successfully in non-fleet mode")
+	} else {
+		// If fleet mode is enabled, the connection to OCM needs to initiate using client ID and client secret
+		// On the managed cluster, the client ID and secret will be fetched from the secret volume however for
+		// local testing, the client ID and secret can be passed directly as flags for ocm-agent CLI.
+		if o.testFleetMode {
```

Comment:
Just a tiny thing. We do not actually need the flag of `test-fleet-mode`, we can just verify if we pass the `--ocm-client-id` and `--ocm-client-secret`, if yes, read it instead of reading the mounted secret(otherwise if we just pass `test-fleet-mode` without `--ocm-client-id` and `--ocm-client-secret`, what gonna happen?).

Reply:
Well, that's a valid point. My intent was to mainly "group" the flags related to testing based on a boolean flag. Actually I see I haven't added the condition for it but basically to have the client ID and secret passed to the `serve` command ONLY while testing since the secret would expose the credential. Else for production clusters it should be accessible only via volume mounts of the secret. Although, its unlikely that someone were to run the `ocm-agent` binary manually on any production cluster so we could still keep it simple based on your suggestion.

Reply:
Removed the testmode flag.